pkg/util/fs_util.go

```diff
@@ -101,13 +106,47 @@ func GetFSFromImage(root string, img v1.Image) error {
 					continue
 				}
 			}
+			if hdr.Typeflag == tar.TypeLink {
+				// If linkname no longer exists, extract hardlink as regular file
+				linkname := filepath.Clean(filepath.Join("/", hdr.Linkname))
+				if CheckWhitelist(linkname) {
+					logrus.Debugf("skipping hardlink from %s to %s because %s is whitelisted", linkname, path, linkname)
+					continue
+				}
+				_, previouslyAdded := fs[linkname]
+				if previouslyAdded || checkWhiteouts(linkname, whiteouts) {
+					if h, ok := hardlinks[linkname]; ok {
+						h.links = append(h.links, hdr)
+						continue
+					}
+					hardlinks[linkname] = &hardlink{
+						links:  []*tar.Header{hdr},
+						reader: tr,
```

Comment:
Hmm, I'd be kinda surprised if saving off the tar reader here works later. Per the docs for `tr.Next`:

> Next advances to the next entry in the tar archive. The Header.Size determines how many bytes can be read for the next file. *Any remaining data in the current file is automatically discarded.*

(emphasis mine)

One problem is that it's basically streaming through the archvie sequentially as it's downloading. It never buffers to disk or RAM, so I don't think you can go "back" to earlier entries in the archive. Tricky.

Reply:
Thanks for pointing this out, fixed.